sky/clouds/service_catalog/data_fetchers/fetch_gcp.py

```diff
@@ -1,579 +1,465 @@
-"""A script that generates Google Cloud GPU catalog.
+"""A script that generates GCP catalog.
 
-Google Cloud does not have an API for querying TPU/GPU offerings, so we crawl
-the information from GCP websites.
+This script uses the GCP APIs to query the list and real-time prices of the
+VMs, GPUs, and TPUs. The script takes about 1-2 minutes to run.
 """
+
 import argparse
+import functools
+import io
+import multiprocessing
 import os
-import re
-from typing import Dict, List, Optional, Tuple
+from typing import Any, Dict, List, Optional
 
-from lxml import html
+import google.auth
+from googleapiclient import discovery
+import numpy as np
 import pandas as pd
-import requests
 
-# pylint: disable=line-too-long
-GCP_VM_PRICING_URL = 'https://cloud.google.com/compute/vm-instance-pricing'
-GCP_VM_ZONES_URL = 'https://cloud.google.com/compute/docs/regions-zones'
-GCP_GPU_PRICING_URL = 'https://cloud.google.com/compute/gpus-pricing'
-GCP_GPU_ZONES_URL = 'https://cloud.google.com/compute/docs/gpus/gpu-regions-zones'
+from sky.adaptors import gcp
+
+# Useful links:
+# GCP SKUs: https://cloud.google.com/skus
+# VM pricing: https://cloud.google.com/compute/vm-instance-pricing
+# GPU pricing: https://cloud.google.com/compute/gpus-pricing
+# TPU pricing: https://cloud.google.com/tpu/pricing
 
-NOT_AVAILABLE_STR = 'Not available in this region'
+# Service IDs found in https://cloud.google.com/skus
+GCE_SERVICE_ID = '6F81-5844-456A'
+TPU_SERVICE_ID = 'E000-3F24-B8AA'
 
-ALL_REGION_PREFIX = ''
-US_REGION_PREFIX = 'us-'
+# The number of digits to round the price to.
+PRICE_ROUNDING = 5
 
 # Refer to: https://github.com/skypilot-org/skypilot/issues/1006
-UNSUPPORTED_VMS = ['t2a-standard', 'f1-micro']
-
-# Supported GPU types and counts.
-# NOTE: GCP officially uses 'A100 40GB' and 'A100 80GB' as the names of the
-# two A100 GPU types. However, in the catalog, we rename them as
-# 'A100' and 'A100-80GB' respectively, for consistency with other clouds.
-GPU_TYPES_TO_COUNTS = {
-    'A100 40GB': [1, 2, 4, 8, 16],
-    'A100 80GB': [1, 2, 4, 8],
-    'T4': [1, 2, 4],
-    'P4': [1, 2, 4],
-    'V100': [1, 2, 4, 8],
-    'P100': [1, 2, 4],
-    'K80': [1, 2, 4, 8],
+# C3 series is still in preview.
+# G2 series has L4 GPU, which is not supported by SkyPilot yet
```

Comment:
Can you say a bit why SkyPilot doesn't support L4?

Reply:
We're hoping to use it to do vicuna inference to save costs.

Reply:
I haven't tried it since the L4 GPU is in preview at the moment. I think the reason is L4 GPUs should be used with g2 VM instances, just like A100 GPUs should be used with A2 instances. SkyPilot does not have that logic.

Reply:
Maybe we can merge the current PR, and have an issue for that?